docs/sources/reference/components/prometheus.remote_write.md

```diff
@@ -472,6 +474,41 @@ If the component falls behind more than one third of the data written since the
 last truncate interval, it is possible for the truncate loop to checkpoint data
 before being pushed to the remote_write endpoint.
 
+### Tuning `max_shards`
+
+The [`queue_config`](#queue_config-block) block allows to configure `max_shards`. The `max_shards` is the maximum
+number of concurrent shards sending samples to the Prometheus-compatible remote write endpoint.
+For each shard, a single remote write request can send up to `max_samples_per_send` samples.
+
+The maximum throughput that Grafana Alloy can achieve when remote writing is a function of
+`max_shards * max_samples_per_send * <write request latency>`. For example, running Alloy with the
+default configuration of 50 `max_shards` and 2000 `max_samples_per_send`, and assuming the
+average latency of a remote write request is 500ms, the maximum throughput achievable is
+about `50 * 2000 * (1s / 500ms) = 200K samples / s`.
```

Comment:
```suggestion
`max_shards * max_samples_per_send * <1 / average write request latency>`. For example, running Alloy with the
default configuration of 50 `max_shards` and 2000 `max_samples_per_send`, and assuming the
average latency of a remote write request is 500ms, the maximum throughput achievable is
about `50 * 2000 * (1s / 500ms) = 200K samples / s`.
```

Reply:
so the goal of this paragraph is to arrive at a reasonable max_shards number, right? Can we give a query which solves for max_shards - something like `max_shards = target_samples_per_sec * avg_latency_in_sec / max_samples_per_send`?